src/bp-members/bp-members-functions.php

```diff
@@ -1629,11 +1631,24 @@ function bp_core_validate_email_address( $user_email ) {
 		}
 	}
 
-	// Is the email already in use?
+	// Is the email already in use in a user?
 	if ( email_exists( $user_email ) ) {
 		$errors['in_use'] = 1;
 	}
 
+	// Is the email already in use in a signup?
+	if ( ! isset( $errors['in_use'] ) ) {
```

Comment:
Multiple pending accounts can be registered with the same email address, but different usernames. That's a bug, IMO.